covalent/executor/base.py

```diff
@@ -65,9 +67,20 @@ def wrapper_fn(
         cb_args = serialized_args.get_deserialized()
         cb_kwargs = serialized_kwargs.get_deserialized()
         retval = cb_fn(*cb_args, **cb_kwargs)
-        if retval_key:
+        if retval_key and retval_key in cb_retvals:
+            if isinstance(cb_retvals[retval_key], list):
+                cb_retvals[retval_key].append(retval)
+            else:
+                cb_retvals[retval_key] = [cb_retvals[retval_key], retval]
```

Comment:
Are we assuming here that no other call_before will have a return value key of `files`? If so, this seems ripe for unexpected behavior; do we know that `cb_retvals[retval_key]` is in fact a path intended for file transfer. If the `files` key is to be used exclusively by File Transfer, we should enforce that at workflow construction.

Reply:
Good points @cjao, currently if there are non-unique retval keys the intention is to accumulate them in an array even if they do not correspond to reserved retval keys (such as `files`) - which would be included in RTD, however I think the way the logic is written would lead to some unexpected behavior if we user's return values are arrays I will revise this and cover it with tests.

I agree that during workflow construction we should check if the user is supplying reserved retval keys and fail hard during workflow construction - will include this.